Sources/SmithyHTTPAPI/Headers.swift

```diff
@@ -138,15 +138,7 @@ public struct Headers: Sendable {
     }
 
     public func exists(name: String) -> Bool {
-        guard headers.index(of: name) != nil else {
-            return false
-        }
-
-        guard let value = value(for: name) else {
-            return false
-        }
-
-        return !value.isEmpty
+        headers.index(of: name) != nil
```

Comment:
Logic modified so that any value on an existing header "exists".  Tests modified below to exercise specific cases.